File: app-miyou/src/main/java/com/miyou/app/domain/monitoring/port/MetricsRollupRepository.java
package com.miyou.app.domain.monitoring.port;

import java.time.Instant;

import com.miyou.app.domain.monitoring.model.MetricsGranularity;
import com.miyou.app.domain.monitoring.model.MetricsRollup;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface MetricsRollupRepository {
	Mono<MetricsRollup> save(MetricsRollup rollup);

	Flux<MetricsRollup> findByGranularityAndBucketStartBetweenOrderByBucketStartAsc(
		MetricsGranularity granularity,
		Instant startTime,
		Instant endTime);
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/monitoring/port/MetricsRollupRepository.kt
package com.miyou.app.domain.monitoring.port

import com.miyou.app.domain.monitoring.model.MetricsGranularity
import com.miyou.app.domain.monitoring.model.MetricsRollup
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.time.Instant

interface MetricsRollupRepository {
    fun save(rollup: MetricsRollup): Mono<MetricsRollup>

    fun findByGranularityAndBucketStartBetweenOrderByBucketStartAsc(
        granularity: MetricsGranularity,
        startTime: Instant,
        endTime: Instant,
    ): Flux<MetricsRollup>
}
